OXRS-SHA-StateMonitor-ESP32-FW.ino

```diff
@@ -68,13 +67,13 @@ OXRS_Input oxrsInput[MCP_COUNT];
 // Ethernet client
 EthernetClient ethernet;
 
-// MQTT client
-PubSubClient mqttClient(MQTT_BROKER, MQTT_PORT, mqttCallback, ethernet);
-OXRS_MQTT mqtt(mqttClient);
-
 // screen functions
 OXRS_LCD screen;

```

Comment:
Do you think we should explicitly pass the `ethernet` pointer into the `screen` constructor? We could then add an overloaded constructor for passing a WiFi pointer for WiFi based devices?

This would probably mean we can remove `show_ethernet()` altogether and just have the IP/MAC display handling inside the `screen.loop()` function?

So basically when you initialise your screen object, you pass either an ethernet or wifi object. And from there on the screen library knows what to use for updating the IP/MAC/linkLED.

Reply:
done